cni/test/e2e-taint/e2e_taint_test.go

```diff
@@ -0,0 +1,256 @@
+	fmt.Println(cmd.String())
+	err = cmd.Run()
+	err = retry.UntilSuccess(func() error {
+		configmap, err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Get(context.TODO(), options.ConfigmapName, metav1.GetOptions{})
+		return err
+	}, retry.Converge(1), retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	configmap, err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Get(context.TODO(), options.ConfigmapName, metav1.GetOptions{})
+	return configmap, err
+}
+func waitForConfigMapDeletion(client client.Interface, name, namespace string) error {
+	err := retry.UntilSuccess(func() error {
+		_, err2 := client.CoreV1().ConfigMaps(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err2 == nil {
+			return fmt.Errorf("configmap %s should be deleted", name)
+		}
+		if errors.IsNotFound(err2) {
+			return nil
+		}
+		return err2
+	}, retry.Converge(1), retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+	return err
+}
+func waitForDaemonSetDeletion(client client.Interface, name, namespace string) error {
+	err := retry.UntilSuccess(func() error {
+		_, err2 := client.AppsV1().DaemonSets(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err2 == nil {
+			return fmt.Errorf("configmap %s should be deleted", name)
+		}
+		if errors.IsNotFound(err2) {
+			return nil
+		}
+		return err2
+	}, retry.Converge(1), retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+	return err
+}
+func TestTaintController(t *testing.T) {
+	type configLocation struct {
+		configmapName      string
+		configmapNamespace string
+		fileLocation       string
+	}
+	type args struct {
+		config             configLocation
+		daemonsetLocations []string
+		namespaces         []v1.Namespace
+	}
+	tests := []struct {
+		name string
+		args args
+	}{{
+		name: "single case",
+		args: args{
+			config:             configLocation{fileLocation: "testdata/singleConfig", configmapNamespace: "default", configmapName: "single"},
+			daemonsetLocations: []string{"testdata/podsConfig/single-critical-label.yaml"},
+			namespaces:         []v1.Namespace{Test1NS},
+		},
+	},
+		{
+			name: "multiple labels in two namespace",
+			args: args{
+				config:             configLocation{fileLocation: "testdata/multiConfig", configmapNamespace: "default", configmapName: "multi"},
+				daemonsetLocations: []string{"testdata/podsConfig/single-critical-label.yaml", "testdata/podsConfig/addon-test2-in-test2.yaml"},
+				namespaces:         []v1.Namespace{Test1NS, Test2NS},
+			},
+		},
+		{
+			name: "multiple labels in one namespace",
+			args: args{
+				config:             configLocation{fileLocation: "testdata/multiInOneConfig", configmapNamespace: "default", configmapName: "mutli.one"},
+				daemonsetLocations: []string{"testdata/podsConfig/single-critical-label.yaml", "testdata/podsConfig/addon-test1.yaml"},
+				namespaces:         []v1.Namespace{Test1NS},
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
When running these I get errors about failed sync and the tests take a long time - seems like something may be wrong

```
/home/howardjohn/go/src/istio.io/istio/cni/test/e2e-taint <nil>
/home/howardjohn/bin/kubectl create configmap single -n default --from-file=/home/howardjohn/go/src/istio.io/istio/cni/test/e2e-taint/testdata/singleConfig/
2020-08-11T21:34:33.313895Z     info    successfully loaded name is critical-test1, namespace is test1,  selector is critical=test1
2020-08-11T21:34:33.322103Z     error   timed out waiting for caches to sync pod controller
=== RUN   TestTaintController/multiple_labels_in_two_namespace
/home/howardjohn/go/src/istio.io/istio/cni/test/e2e-taint <nil>
/home/howardjohn/bin/kubectl create configmap multi -n default --from-file=/home/howardjohn/go/src/istio.io/istio/cni/test/e2e-taint/testdata/multiConfig/
2020-08-11T21:34:38.535002Z     info    successfully loaded name is critical-test1, namespace is test1,  selector is critical=test1
2020-08-11T21:34:38.535037Z     info    successfully loaded name is addon=test2, namespace is test2,  selector is addon=test2
2020-08-11T21:34:38.542615Z     error   timed out waiting for caches to sync pod controller
=== RUN   TestTaintController/multiple_labels_in_one_namespace
/home/howardjohn/go/src/istio.io/istio/cni/test/e2e-taint <nil>
/home/howardjohn/bin/kubectl create configmap mutli.one -n default --from-file=/home/howardjohn/go/src/istio.io/istio/cni/test/e2e-taint/testdata/multiInOneConfig/
2020-08-11T21:34:49.203837Z     info    successfully loaded name is multi-in-one-test, namespace is test1,  selector is critical=test1
2020-08-11T21:34:49.203924Z     info    successfully loaded name is addon=test2, namespace is test1,  selector is addon=test2
2020-08-11T21:34:49.213645Z     error   timed out waiting for caches to sync pod controller
--- PASS: TestTaintController (21.22s)
    --- PASS: TestTaintController/single_case (5.20s)
    --- PASS: TestTaintController/multiple_labels_in_two_namespace (10.60s)
    --- PASS: TestTaintController/multiple_labels_in_one_namespace (5.41s)
```

Reply:
resolved cache sync issues